operator/validator/event_handler.go

```diff
@@ -157,15 +156,24 @@ func (c *controller) handleValidatorAddedEvent(
 		return nil, errors.Wrap(err, "failed to get nonce")
 	}
 
-	err = c.eventHandler.SaveEventData(event.TxHash)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to save event data")
-	}
+	var valid, malformed bool
 
-	err = c.eventHandler.BumpNonce(event.Owner)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to bump the nonce")
-	}
+	defer func() {
+
+		if valid || malformed {
```

Comment:
Why do we need this condition, shouldn't we always save & bump?

Reply:
no, that's the point of this PR, only save and bump when the event is malformed or valid.
if for example we saved+bumped and then had a crash/db error in the middle, we wont to re-run this event.
if at any point we have a "system error"  we should re-run the whole thing.

Like we talked, this is unlikely to happen and if we have DB/system error we should probably quit/crash. but for example in our stage env there are a lot of redeploys, if we were redeployed during event handling we would just re-run it! saving the saving the event and bumping the nocne before finishing might result in bad data in this case.

@olegshmuelov please review as well